crates/common/tedge_utils/src/file.rs

```diff
@@ -359,6 +360,7 @@ pub async fn overwrite_file(file: &Path, content: &str) -> Result<(), FileError>
                     from: e,
                 })
                 .await?;
+            f.flush().await?;
```

Comment:
The current sequence of calls (`sync_all`, `write_all`, `flush`) has to be rewritten as  (`write_all`, `flush`, `sync_all`)